lib/ReactViews/Custom/Chart/MomentPointsChart.jsx

```diff
@@ -31,36 +33,37 @@ class MomentPointsChart extends React.Component {
         domain: basisItemScales.y.domain(),
         range: scales.y.domain()
       });
-      const interpolatedPoints = chartItem.points.map(p =>
-        interpolate(p, basisItem.points, basisToSourceScale)
-      );
+      const interpolatedPoints = chartItem.points.map(p => ({
+        ...p,
+        ...interpolate(p, basisItem.points, basisToSourceScale)
+      }));
       return interpolatedPoints;
     }
     return chartItem.points;
   }
 
   doZoom(scales) {
-    const points = this.props.chartItem.points;
-    if (points.length === 0) return;
-    const n = points.length;
-    const spacing = (scales.x(points[n - 1].x) - scales.x(points[0].x)) / n;
-    const dots = document.querySelectorAll(`g#${this.props.id} circle`);
-    dots.forEach((dot, i) => {
+    const points = this.points;
+    if (points.length === 0) {
```

Comment:
could points be undefined? ie if the chartItem from props didn't have a points key?

Reply:
`chartItem` is [defined here](https://github.com/TerriaJS/terriajs/blob/9e6a18f8c656701f2cad8d550d32da468f8eccc0/lib/ModelMixins/ChartableMixin.ts#L60) as array. And because all catalog items are `.ts` we can be confident that this will always be defined.